docs/pages/reference/build/stages_images.md

```diff
@@ -0,0 +1,107 @@
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vRKB-_Re-ZhkUSB45jF9GcM-3gnE2snMjTOEIQZSyXUniNHKK-eCQl8jw3tHFF-a6JLAr2sV73lGAdw/pub?w=821&amp;h=362" >
+</a>
+</div>
+
+<div id="artifact" class="tabcontent">
+<a class="google-drawings" href="https://docs.google.com/drawings/d/e/2PACX-1vRD-K_z7KEoliEVT4GpTekCkeaFMbSPWZpZkyTDms4XLeJAWEnnj4EeAxsdwnU3OtSW_vuKxDaaFLgD/pub?w=1800&amp;h=850" data-featherlight="image">
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vRD-K_z7KEoliEVT4GpTekCkeaFMbSPWZpZkyTDms4XLeJAWEnnj4EeAxsdwnU3OtSW_vuKxDaaFLgD/pub?w=640&amp;h=301">
+</a>
+</div>
+
+<!-- 301 -->
+
+**All works with _stages_ are done by werf, and you only need to write config correctly.**
+
+Each _stage_ is assembled in an ***assembly container*** based on an image of the previous _stage_. The result of the assembly _stage_ and _stage conveyor_, in general, is the ***stages cache***: each _stage_ relates to one Docker image.
+
+The _signature_ is calculated for every _stage_ at each build. At the last step of the build when saving _stages cache_, the _signature_ is used for tagging stage (read [how stage images are named](#how-stage-images-are-named)). This logic allows to assembly only _stages_ whose the _stages cache_ does not exist in a [stages storage](#what-is-a-stages-storage) and use a cache for re-assembly stages.
+
+<div class="rsc" markdown="1">
+
+<div class="rsc-description" markdown="1">
+
+  The ***stage signature*** is the checksum of _stage dependencies_ and previous _stage signature_. In the absence of _stage dependencies_, the _stage_ is skipped.
+
+  It means that the _stage conveyor_, e.g., image _stage conveyor_, can be reduced to several _stages_ or even to single _from_ stage.
+
+</div>
+
+<div class="rsc-example">
+<a class="google-drawings" href="https://docs.google.com/drawings/d/e/2PACX-1vR6qxP5dbQNlHXik0jCvEcKZS2gKbdNmbFa8XIem8pixSHSGvmL1n7rpuuQv64YWl48wLXfpwbLQEG_/pub?w=572&amp;h=577" data-featherlight="image">
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vR6qxP5dbQNlHXik0jCvEcKZS2gKbdNmbFa8XIem8pixSHSGvmL1n7rpuuQv64YWl48wLXfpwbLQEG_/pub?w=286&amp;h=288">
+</a>
+</div>
+
+</div>
+
+<div style="clear: both;"></div>
+
+### How stage images are named?
+
+A stage image name depends on where the stage is stored. At the moment, stages can be stored only in a local stages storage (read more about [stages storage](#what-is-a-stages-storage)). So, each stage can be named only by the following schema — `werf-stages-storage/PROJECT_NAME:SIGNATURE`.
+
+Each stage image is tagged by werf using a _stage signature_.
+
+The project name is used in every stage image name, so for every project there is an independent stages cache.
+
+## What is a stages storage?
+
+Stages storage is a place where all stages of a project are stored. Many commands use stages storage and require a `--stages-storage` option.
```

Comment:
Stages storage keeps project stages. Stages can be stored in Docker Repo or locally, on a host machine. At the moment, only local storage, `:local`, is supported, 

Most commands use stages and require specified stages storage, defined by `--stages-storage` option or `$WERF_STAGES_STORAGE`.